core/views.py

```diff
@@ -22,105 +22,141 @@
+            return match_
+    return pd.DataFrame()
+
+@timeit
+def find_address_match(cms_metric_df, place_address):
+
+    place_address = format_address(place_address)
+
+    # Dict look ups are faster than iterating through the df
+    addresses = dict(zip(cms_metric_df['Address'].values,cms_metric_df['Address'].index))
+
+    if place_address in addresses:
+        match_ = cms_metric_df.loc[addresses[place_address],:]
+        print(f"address match: '{place_address}' with '{match_['Address']}'")
+        return match_
+    else:
+        # Try closest match
+        matches = get_close_matches(place_address, addresses.keys(), n=1, cutoff=0.8)
+        if matches:
+            match_ = cms_metric_df.loc[addresses[matches[0]],:]
+            print(f"address match: '{place_address}' with '{match_['Address']}'")
+            return match_
+
     #return empty df if no hits
     return pd.DataFrame()
 
-def expand_address_abbreviations(place_address):
-    place_address = place_address.replace(' ave', ' avenue')
-    place_address = place_address.replace(' rd', ' road')
-    place_address = place_address.replace(' st', ' street')
-    place_address = place_address.replace(' dr', ' drive')
-    place_address = place_address.replace(' ct', ' court')
-    place_address = place_address.replace(' e ', ' east')
-    place_address = place_address.replace(' w ', ' west')
-    place_address = place_address.replace(' n ', ' north')
-    place_address = place_address.replace(' s ', ' south')
+@timeit
+def format_address(place_address):
+    print("place_address preformat: ", place_address)
+    place_address = place_address.lower().replace("-", " ").replace("/", " ")
+    place_address = place_address.replace(' ave ', ' avenue ')
+    place_address = place_address.replace(' rd ', ' road ')
+    place_address = place_address.replace(' st ', ' street ')
+    place_address = place_address.replace(' dr ', ' drive ')
+    place_address = place_address.replace(' ct ', ' court ')
+    place_address = place_address.replace(' e ', ' east ')
+    place_address = place_address.replace(' w ', ' west ')
+    place_address = place_address.replace(' n ', ' north ')
+    place_address = place_address.replace(' s ', ' south ')
+    place_address = place_address.replace(' ne ', ' northeast ')
+    place_address = place_address.replace(' nw ', ' northwest ')
+    place_address = place_address.replace(' se ', ' southeast ')
+    place_address = place_address.replace(' sw ', ' southwest ')
+    place_address = place_address.replace(' blvd ', ' boulevard ')
+    place_address = place_address.replace(' pkwy ', ' parkway ')
+    place_address = place_address.replace(' pl ', ' place ')
+    place_address = place_address.replace(' ln ', ' lane ')
+    place_address = place_address.replace(' hwy ', ' highway ')
+    place_address = place_address.strip()
+    place_address = place_address.replace(", united states", "")
```

Comment:
what issue was this solving?

Reply:
words avenue where getting turned aveavenue.

The formatted addresses from gmaps have united states on the end. 

Will break the address changes out into there own PR